modules/clickhouse/clickhouse.go

```diff
@@ -0,0 +1,179 @@
+// as a command line argument to the container.
+func WithConfigFile(configFile string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		cf := testcontainers.ContainerFile{
+			HostFilePath:      configFile,
+			ContainerFilePath: "/etc/clickhouse-server/config.d/config.xml",
+			FileMode:          0755,
+		}
+		req.Files = append(req.Files, cf)
+	}
+}
+
+// WithConfigFile sets the YAML config file to be used for the clickhouse container
+// It will also set the "configFile" parameter to the path of the config file
+// as a command line argument to the container.
+func WithYamlConfigFile(configFile string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		cf := testcontainers.ContainerFile{
+			HostFilePath:      configFile,
+			ContainerFilePath: "/etc/clickhouse-server/config.d/config.yaml",
+			FileMode:          0755,
+		}
+		req.Files = append(req.Files, cf)
+	}
+}
+
+// WithDatabase sets the initial database to be created when the container starts
+// It can be used to define a different name for the default database that is created when the image is first started.
+// If it is not specified, then the default value("clickhouse") will be used.
+func WithDatabase(dbName string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Env["CLICKHOUSE_DB"] = dbName
+	}
+}
+
+// WithPassword sets the initial password of the user to be created when the container starts
+// It is required for you to use the ClickHouse image. It must not be empty or undefined.
+// This environment variable sets the password for ClickHouse.
+func WithPassword(password string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		req.Env["CLICKHOUSE_PASSWORD"] = password
+	}
+}
+
+// WithUsername sets the initial username to be created when the container starts
+// It is used in conjunction with WithPassword to set a user and its password.
+// It will create the specified user with superuser power.
+// If it is not specified, then the default user of clickhouse will be used.
+func WithUsername(user string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		if user == "" {
+			user = defaultUser
+		}
+
+		req.Env["CLICKHOUSE_USER"] = user
+	}
+}
+
+// RunContainer creates an instance of the ClickHouse container type
+func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*ClickHouseContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image: defaultImage,
+		Env: map[string]string{
+			"CLICKHOUSE_USER":                      defaultUser,
+			"CLICKHOUSE_PASSWORD":                  defaultUser,
+			"CLICKHOUSE_DB":                        defaultDatabaseName,
+			"CLICKHOUSE_DEFAULT_ACCESS_MANAGEMENT": "1",
```

Comment:
Do you know what is the effect of this? I can not find any in the [docs](https://hub.docker.com/r/clickhouse/clickhouse-server/). I did some tests in java and it is like has no effect.

Reply:
Which part did you ask for? All of these env variables is written in [docs](https://hub.docker.com/r/clickhouse/clickhouse-server/) you linked.

Reply:
I found the [PR](https://github.com/ClickHouse/ClickHouse/pull/20139). I think the env var should be set by the user if needed and not update the default value provided by the image.

Reply:
Ah, I guess I misunderstand purpose of this variable acording to [docs](https://hub.docker.com/r/clickhouse/clickhouse-server/). You are right this setting does not needed to set as default.